project/src/pages/room/room.tsx

```diff
@@ -1,144 +1,122 @@
-import { nearPlacesOffers } from '../../mocks/offers';
-import { reviews } from '../../mocks/reviews';
+import { useParams } from 'react-router-dom';
 import { useState } from 'react';
+import { useEffect } from 'react';
+import { Offer } from '../../types/offer';
+import { AuthorizationStatus } from '../../const';
+import { useAppDispatch, useAppSelector } from '../../hooks';
+import { fetchNearPlacesOffersAction, fetchReviewsAction } from '../../store/api-actions';
+import NotFound from '../../pages/not-found/not-found';
 import Header from '../../components/header/header';
 import ReviewForm from '../../components/review-form/review-form';
 import ReviewList from '../../components/review-list/review-list';
 import Map from '../../components/map/map';
 import Offers from '../../components/offers/offers';
-import { CITIES } from '../../const';
 
 function Room(): JSX.Element {
   const [activeCard, setActiveCard] = useState<number | null>(null);
 
+  const dispatch = useAppDispatch();
+
+  const offerId = Number(useParams().id);
+
+  useEffect(() => {
+    dispatch(fetchNearPlacesOffersAction(offerId));
+    dispatch(fetchReviewsAction(offerId));
+  }, [dispatch, offerId]);
+
+  const offers = useAppSelector((state) => state.offers);
+  const offer: Offer | undefined = offers.find((element) => element.id === offerId);
```

Comment:
```suggestion
  const offer = offers.find((element) => element.id === offerId);
```

итак понятно что там будет. ТС умный)))